lighthouse-cli/test/fixtures/static-server.js

```diff
@@ -21,136 +21,177 @@ const HEADER_SAFELIST = new Set(['x-robots-tag', 'link']);
-        statusCode = parseInt(params.get('status_code'), 10);
-      }
+  _requestHandler(request, response) {
+    const requestUrl = parseURL(request.url);
+    const filePath = requestUrl.pathname;
+    const queryString = requestUrl.search && parseQueryString(requestUrl.search.slice(1));
+    let absoluteFilePath = path.join(__dirname, filePath);
+
+    const sendResponse = (statusCode, data) => {
+      // Used by Smokerider.
+      if (this._dataTransformer) data = this._dataTransformer(data);
+
+      const headers = {'Access-Control-Allow-Origin': '*'};
+
+      const contentType = mime.lookup(filePath);
+      const charset = mime.lookup(contentType);
+      // `mime.contentType` appends the correct charset too.
+      // Note: it seems to miss just one case, svg. Doesn't matter much, we'll just allow
+      // svgs to fallback to binary encoding. `Content-Type: image/svg+xml` is sufficient for our use case.
+      // see https://github.com/jshttp/mime-types/issues/66
+      if (contentType) headers['Content-Type'] = mime.contentType(contentType);
+
+      let delay = 0;
+      let useGzip = false;
+      if (queryString) {
+        const params = new URLSearchParams(queryString);
+        // set document status-code
+        if (params.has('status_code')) {
+          statusCode = parseInt(params.get('status_code'), 10);
+        }
 
-      // set delay of request when present
-      if (params.has('delay')) {
-        delay = parseInt(params.get('delay'), 10) || 2000;
-      }
+        // set delay of request when present
+        if (params.has('delay')) {
+          delay = parseInt(params.get('delay'), 10) || 2000;
+        }
 
-      if (params.has('extra_header')) {
-        const extraHeaders = new URLSearchParams(params.get('extra_header'));
-        for (const [headerName, headerValue] of extraHeaders) {
-          if (HEADER_SAFELIST.has(headerName.toLowerCase())) {
-            headers[headerName] = headers[headerName] || [];
-            headers[headerName].push(headerValue);
+        if (params.has('extra_header')) {
+          const extraHeaders = new URLSearchParams(params.get('extra_header'));
+          for (const [headerName, headerValue] of extraHeaders) {
+            if (HEADER_SAFELIST.has(headerName.toLowerCase())) {
+              headers[headerName] = headers[headerName] || [];
+              headers[headerName].push(headerValue);
+            }
           }
         }
+
+        if (params.has('gzip')) {
+          useGzip = Boolean(params.get('gzip'));
+        }
+
+        // redirect url to new url if present
+        if (params.has('redirect')) {
+          return setTimeout(sendRedirect, delay, params.get('redirect'));
+        }
       }
 
-      if (params.has('gzip')) {
-        useGzip = Boolean(params.get('gzip'));
+      if (useGzip) {
+        data = zlib.gzipSync(data);
+        headers['Content-Encoding'] = 'gzip';
+        // Set special header for Lightrider, needed for Smokerider.
```

Comment:
this confuses me, if LR is fetching this won't it create this header on its own?

AFAIK the `X-TotalFetchedSize` header is necessary because LR decompresses the content and the transferSize to Chrome is the uncompressed size instead. If we're serving gzip'd content, it feels like this header logic should live alongside whatever does the decompression.

Reply:
fair question :) Only the specific fetcher we use in production adds this header. The LR smoketests will use a barebones fetcher that is just the local network, so it won't have this header that LH expects.

it _could_ be done just in the LR test as part of setup, but would require a way to add headers. at that point we might as well pivot to express ...

Reply:
> The LR smoketests will use a barebones fetcher that is just the local network

Ah I see and this barebones fetcher still has the production behavior of masking the compression sizes that we don't have the ability to modify (or add the header too)?

Could we add this note to the comment here then? Maybe just "LR only adds this header in production mode, but the test fetcher still masks transferSize beyond our control."

Reply:
No to the masking bit - the size info is readily available. The idea is to still exercise this code path in LH / Smokerider. Updated comment.

Reply:
gotcha thanks!